stdlib/public/Concurrency/AsyncStream.swift

```diff
@@ -167,7 +265,39 @@ public struct AsyncStream<Element> {
     build(Continuation(storage: storage))
   }
 
-  
+  /// Constructs an asynchronous stream from a given element-producing
+  /// closure, with an optional closure to handle cancellation.
+  ///
+  /// - Parameters:
+  ///   - produce: A closure that asynchronously produces elements for the
+  ///    stream.
+  ///   - onCancel: A closure to execute when cancelling the stream's task.
+  ///
+  /// Use this convenience initializer when you have an asychronous function
+  /// that can produce elements for the stream, and don't want to invoke
+  /// a continuation manually. This initializer "unfolds" your closure into
+  /// a full-blown asynchronous stream. The created stream handles adherence to
+  /// the `AsyncSequence` protocol automatically, including termination (whether
+  /// by cancellation or by returning `nil` from the closure to finish
+  /// iteration).
+  ///
+  /// The following example shows an `AsyncStream` created with this
+  /// initializer that produces random numbers on a one-second interval.
+  ///
+  ///     let stream = AsyncStream<Int>(
+  ///         unfolding: {
+  ///             await Task.sleep(1 * 1_000_000_000)
+  ///             return Int.random(in: 1...10)
+  ///         },
+  ///         onCancel: { @Sendable () in print ("Canceled.") }
+  ///     )
```

Comment:
Can/should this use the [multiple trailing closure syntax](https://github.com/apple/swift-evolution/blob/main/proposals/0279-multiple-trailing-closures.md#proposed-solution) from Swift 5.3?